### DISCUSSION TITLE
Farm vs Vite with Rolldown

### DISCUSSION THREAD

**Discussion by martinszeltins** [farm-fe/farm#1486] [General]
What are your thoughts about Vite with Rolldown that will be released later this year? It will basically do everything that Farm does so does it mean that there will be no real need for Farm anymore? What are your thoughts?

Rolldown will address all the issues that Farm is trying to address.

**Comment by wre232114** [comment]
Hi, here are my thoughts:
1. Farm's Vite compatibility is only for making migrate existing projects to Farm easily, we'll focus on Farm rust plugin ecosystem in the future, there is a set of usefully rust plugins here https://github.com/farm-fe/plugins/tree/main/packages
2. Rolldown cannot address all the issues of Vite in a short time, Rolldown is only used to replace esbuild currently. To address all issues, Vite should re-design its architecture while being compatible with old versions, this could be a long run.

**Reply by martinszeltins** [reply]
According to this, Rolldown will be replacing both `esbuild` and `Rollup` for Vite - https://rolldown.rs/about

Wouldn't that mean it will be the same as Farm?

**Reply by wre232114** [reply]
No, after replacing esbuild and rollup, there is still a lot of work to do to resolve all the issues.

**Reply by martinszeltins** [reply]
What are those issues?

**Reply by wre232114** [reply]
`consistency` and `huge humber of requests during development`, Vite is unbundled during development and bundle in production, that's the key issue. To address those issues, I think vite should refactor its architecture
